js/gracelib.js

```diff
@@ -1346,15 +1351,32 @@ var stdout = Grace_allocObject();
 stdout.methods.write = function(junk, s) {
     minigrace.stdout_write(s._value);
 }
+stdout.methods.isatty = function() {
+        if(typeof(process) != "undefined") {
```

Comment:
You are uniformly using ```typeof(process) != "undefined"``` to decide if we are in a browser or in Node.js   I was using a similar test on "global", which is defined by node but not in the browser.   I wonder if we should make these uniforms, and introduce functions ```inBrowser``` and ```inNode```  to encapsulate them.  Maybe it's fine as is, because the code that looks at global goes on to index into global, while the code that looks at process goes on to use process.